_posts/2024-07-24-pytorch2-4.md

```diff
@@ -0,0 +1,155 @@
+
+The workflow is as explained in the AOTInductor [tutorial](https://pytorch.org/docs/main/torch.compiler_aot_inductor.html), in addition to that users could now add the freezing flag to get better performance:
+```
+export TORCHINDUCTOR_FREEZING=1
+```
+
+
+### [Beta] New Higher-level Python Custom Operator API
+
+We’ve added a new higher-level Python Custom Operator API that makes it easier than before to extend PyTorch with custom operators that behave like PyTorch’s built-in operators. Operators registered using the [new high-level torch.library APIs](https://pytorch.org/docs/2.4/library.html#module-torch.library) are guaranteed to be compatible with `torch.compile` and other PyTorch subsystems; authoring a custom operator in Python using the previous [low-level torch.library APIs](https://pytorch.org/docs/2.4/library.html#low-level-apis)required deep understanding of PyTorch internals and has many footguns.
+
+Please see the [tutorial](https://pytorch.org/tutorials/advanced/python_custom_ops.html) for more information.
+
+### [Beta] Switching TCPStore’s default server backend to _libuv_
+
+Introduced a new default server backend for TCPStore built with `libuv` which should introduce significantly lower initialization times and better scalability. This should ideally benefit users with a much shorter startup time when accounting for large-scale jobs.
+
+For more information on the motivation + fallback instructions please refer to this [tutorial](https://pytorch.org/tutorials/intermediate/TCPStore_libuv_backend.html).
+
+
+## Prototype Features
+
+
+### [PROTOTYPE] FSDP2: DTensor-based per-parameter-sharding FSDP
+
+FSDP2 is a new fully sharded data parallelism implementation that uses dim-0 per-parameter sharding to resolve fundamental composability challenges with FSDP1's flat-parameter sharding.
+
+For more information regarding the motivation / design for FSDP2 please refer to the [RFC on Github](https://github.com/pytorch/pytorch/issues/114299).
+
+
+### [PROTOTYPE] _torch.distributed.pipelining_, simplified pipeline parallelism
+
+Pipeline Parallelism is one of the primitive parallelism techniques for deep learning. It allows the execution of a model to be partitioned such that multiple micro-batches can execute different parts of the model code concurrently.
+
+`torch.distributed.pipelining` provides a toolkit that allows for easy implementation of pipeline parallelism on general models while also offering composability with other common PyTorch distributed features like DDP, FSDP, or tensor parallel.
+
+For more information on this please refer to our [documentation](https://pytorch.org/docs/main/notes/get_start_xpu.html) and [tutorial](https://pytorch.org/tutorials/intermediate/pipelining_tutorial.html).
```

Comment:
have the wrong link for documentation: https://pytorch.org/docs/main/distributed.pipelining.htm

Thanks for adding the blurb though!

Reply:
Updated